add sd-jwt vcld (#459)
openid/OpenID4VP: openid-4-verifiable-presentations-1_0.md

{backmatter}

<reference anchor="JSON-LD" target="https://www.w3.org/TR/json-ld11/">
  <front>

>     <title>Verifiable Credentials Data Model 1.1</title>

Reviewer: Oh, Danny, this isn't VCDM. Is this VCDM? This isn't VCDM, is it? I didn't think so.

```suggestion
    <title>JSON-LD 1.1</title>
```

<{{examples/response/kb_jwt_unsecured.json}}

### SD-JWT VCDM {#sd-jwt_vcdm}

> SD-JWT VCDM (Verifiable Credential Data Model) uses the IETF SD-JWT VC [@!I-D.ietf-oauth-sd-jwt-vc] credential format and allows to incorporate existing data data models, e.g., W3C VCDM [@?VC_DATA], while enabling a consistent approach to selective disclosure.

Reviewer 2: ```suggestion
SD-JWT VCDM (Verifiable Credential Data Model) uses the IETF SD-JWT VC [@!I-D.ietf-oauth-sd-jwt-vc] credential format and allows to incorporate existing data models that use Linked Data, e.g., W3C VCDM [@?VC_DATA], while enabling a consistent approach to selective disclosure.
```

Reviewer 3: should not only limited to data models which use linked data; Selective disclosure in W3CVDM is also consistent. "efficient" approach might be bettr

> SD-JWT VCDM (Verifiable Credential Data Model) uses the IETF SD-JWT VC [@!I-D.ietf-oauth-sd-jwt-vc] credential format and allows to incorporate existing data data models, e.g., W3C VCDM [@?VC_DATA], while enabling a consistent approach to selective disclosure.

Reviewer 3: please change to "SD-JWT VCDM (Verifiable Credential Data Model) uses the IETF SD-JWT VC [@!I-D.ietf-oauth-sd-jwt-vc] credential format and allows to incorporate existing data data models especially those that use linked data, e.g., W3C VCDM [@?VC_DATA], while enabling ann efficient approach to selective disclosure.

Reviewer 4: "that use Linked Data" was added in the meantime, so I assume that this has been addressed.

Information contained in SD-JWT VCDM credentials can be processed using a JSON-LD [@?JSON-LD] processor after applying SD-JWT VC processing.

> When IETF SD-JWT VC is mentioned in this specification, SD-JWT VCDM defined in this section MAY be used.

Reviewer 3: SD-JWT VCDM must be used in case W3C VC Data Model or linked data shall be used

Reviewer 2: when we discussed previously in the WG, this is the scope of work the WG agreed on (from #445): "To expand the number of use-case, define how to combine sd-jwt vc and json-ld based payload in way that uses sd-jwt vc as a securing mechanism, and allows to used w3c vcdm claims for business processing **(compliance to w3c vcdm is not required)**". which is why w3c vcdm compliance is not required.

Reviewer 4: @awoie When this discussion is resolved, I suggest to merging this sentence with the paragraph that I proposed above "SD-JWT VCDM credentials are SD-JWT VC credentials..."

Reviewer 5: Agree with @Sakurann that it makes sense that the goal is enabling JSON-LD payloads, rather than (incorrectly) claiming conformance with W3C VCDM.

Implementers of SD-JWT VCDM MUST use valid values for the `vct` Claim defined in IETF SD-JWT VC [@!I-D.ietf-oauth-sd-jwt-vc].

For backward compatibility with JWT processors, the following registered JWT claims MUST be used:

* To represent the validity period of SD-JWT VCDM (i.e., cryptographic signature), `exp`/`nbf`/`iat` Claims encoded as a UNIX timestamp (NumericDate) MUST be used.
* `iss` Claim MUST represent the Credential Issuer.
* `status` Claim MUST represent the information to obtain the status of the Credential.
* `sub` Claim MUST represent the subject identifier of the Credential.

IETF SD-JWT VC is extended with the following claims:

* `ld`: OPTIONAL. Contains a JSON-LD [@!JSON-LD] object in compact form, e.g., [@?VC_DATA].

The following outlines a suggested non-normative set of processing steps for SD-JWT VCDM:

> 1. SD-JWT VC Processing:

Reviewer 2: ```suggestion
1. SD-JWT VC Processing:

```

<{{examples/response/kb_jwt_unsecured.json}}

### SD-JWT VCLD {#sd-jwt_vcld}

SD-JWT VCLD (SD-JWT Verifiable Credentials with JSON-LD) extends the IETF SD-JWT VC [@!I-D.ietf-oauth-sd-jwt-vc] credential format and allows to incorporate existing data models that use Linked Data, e.g., W3C VCDM [@?VC_DATA], while enabling a consistent and uncomplicated approach to selective disclosure.

> 

Reviewer 4: I don't understand (there seems to be a word missing in your first sentence?). This extension is specifically targeted towards existing data models that use linked data - most other data models can just be used as they are, they don't need additional guidance.

> 

Reviewer 3: @danielfett you currently only include JSON-LD into SD-JWT VC but you ignore seemingly the use of atomic credentials as possible in W3CVCDM but impossible in SD-JWT VC. Only JSON-LD is great but not sufficent

> 

Reviewer 6: List of requirements was presented to the WG in the meeting before the oauth2 security workshop and the list did include W3CVCDM v1.1 and ETSI TS 119 472-1 and additional requirements known from the use cases.

These requirements were also shared with the authors months ago. Ignoring them doesn't mean requirements weren't shared. Furthermore, the requirements have been shared by some of the member states and the proposal was done 1.5 years ago.

> 

Reviewer 6: We even tried to compare the different data models and their capabilities:
https://docs.google.com/spreadsheets/d/1zAcOlB0BNRlEn-doJEew01XJtflU9q3ODVQBd8VuJyk/edit?usp=sharing

> 

Reviewer 7: The authors have very little power in the OIDF process - if you have requirements, you need to share them with the working group.

I missed that meeting unfortunately, so thanks for providing that context. I agree these were shared at that meeting - the minuted agreement from that meeting is:

> the scope of work the WG agreed on: "To expand the number of use-case, define how to combine sd-jwt vc and json-ld based payload in way that uses sd-jwt vc as a securing mechanism, and allows to used w3c vcdm claims for business processing (compliance to w3c vcdm is not required)"

I still do not see any agreement that micro credentials would be included.

If you have a document that explains what use cases they're needed for when the credential format already supports selective disclosure that would be super helpful.

> 

Reviewer 6: It is up to the working group to decide what to do. If the model cannot meet the use case requirements, the WG shouldn't expect from use cases to use the model.

> 

Reviewer 4: To the original point:

> W3VCDM contains also consistent Selective disclosure. 

I assume you refer to section 8.5 of VCDM 1.1. Clearly, VCDM 1.1 supports selective disclosure via microcredentials. OpenID4VP already supports those through the existing VCDM 1.1 credential format in Section B.1. The purpose of this PR is to additionally enable the capability to use SD-JWT VC as a "signature scheme that allows for selective disclosure" (section 8.5).

> 

Reviewer 5: Regarding VCs and selective disclosure:
- https://github.com/danielfett/sd-jwt-vc-dm/issues/2
- https://www.w3.org/TR/vc-di-ecdsa/#ecdsa-sd-2023
- https://www.w3.org/TR/vc-jose-cose/#securing-with-sd-jwt

No SD-JWT VC needed for selective disclosure for VCs.

> 

Reviewer 3: @jogu As @alenhorvat mentioned the requirements were shared. As given in the minutes: https://docs.google.com/document/d/1y3milcqMkAHqf4862ANoCA7irg6gNaDI3aNOVaZMoVY/edit?tab=t.0#heading=h.xzdvf9ioflwg

"-	“The goal is to add W3C VCDM data model and processing to this defined credential - secured by SD JWT”

Exactly this is not achieved with current spec. Other minutes were WG decided to change this are not present, Means the current specification exactly does not reflect the agreed requirements as parts of VCDM not useable.

Information contained in SD-JWT VCLD credentials can be processed using a JSON-LD [@?JSON-LD] processor after the SD-JWT VC processing.

When IETF SD-JWT VC is mentioned in this specification, SD-JWT VCLD defined in this section MAY be used.

#### Format

> SD-JWT VCLD credentials are valid SD-JWT VC credentials and all requirements from [@!I-D.ietf-oauth-sd-jwt-vc] apply. Additionally, the requirements listed in this section apply.

Reviewer 4: This text does not exclude additional requirements to be defined. But the existing requirements cannot be relaxed/ignored.

> SD-JWT VCLD credentials are valid SD-JWT VC credentials and all requirements from [@!I-D.ietf-oauth-sd-jwt-vc] apply. Additionally, the requirements listed in this section apply.

Reviewer 3: not all requirements in SD-JWT VC are mandatory. So to be precise it shall be "SD-JWT VCLD credentials are valid SD-JWT VC credentials and all mandatory requirements..."

As spec currently only defines how to integrate linked data it`s bit too less --> atomic credentials resp. integration data model missed

> SD-JWT VCLD credentials are valid SD-JWT VC credentials and all requirements from [@!I-D.ietf-oauth-sd-jwt-vc] apply. Additionally, the requirements listed in this section apply.

Reviewer 4: I don't understand. All requirements apply. The MUSTs are MUSTs, the SHOULDs are SHOULDs, and so on, as SD-JWT VCLD credentials are valid SD-JWT VCs. Why would only the MUSTs and MUST NOTs apply?

> SD-JWT VCLD credentials are valid SD-JWT VC credentials and all requirements from [@!I-D.ietf-oauth-sd-jwt-vc] apply. Additionally, the requirements listed in this section apply.

Reviewer 3: Recommend to change the spec with following focus

- When W3CVCDM or SD-JWT VC define a property/claim in exactly same way (semantics/features), the claim defined by SD_JWT VC or RFC 7519 apply

- If both so VCDM v2.0 or IETF SD-JWT VC , define a property/claim with similar purpose but different capabilities, then the spec shall allow a choice between the property defined in W3CVCDM and the claim defined in either SD-JWT VC  

- If VCDM v2.0  defines a property not covered by anything in IETF SD-JWT VC then the spec shall define such a property

would give flexibility while staying within SD-JWT VC

> SD-JWT VCLD credentials are valid SD-JWT VC credentials and all requirements from [@!I-D.ietf-oauth-sd-jwt-vc] apply. Additionally, the requirements listed in this section apply.

Reviewer 4: That would create a completely new format in the case there are collisions (second case you listed). The property that every SD-JWT VC DM/LD credential should also be a valid SD-JWT VC credential was pretty much a consensus from the beginning.

> SD-JWT VCLD credentials are valid SD-JWT VC credentials and all requirements from [@!I-D.ietf-oauth-sd-jwt-vc] apply. Additionally, the requirements listed in this section apply.

Reviewer 5: In my opinion the only way how this could possibly (remotely) work is if there is no ambiguity or choice whatsoever. For every feature (issuer, type, timestamps, expiration, status list, proofs, schemas, etc.), decide whether you want to use the JWT feature or the W3C feature.

> SD-JWT VCLD credentials are valid SD-JWT VC credentials and all requirements from [@!I-D.ietf-oauth-sd-jwt-vc] apply. Additionally, the requirements listed in this section apply.

Reviewer 3: @danielfett "The property that every SD-JWT VC DM/LD credential should also be a valid SD-JWT VC credential was pretty much a consensus from the beginning." --> with the pre-condition that the agreed properties like LinkedData and Atomic Credentials work. Which is seemingly not the case for Atomic Credentials now

Means the requirements as agreed in the consensus are NOT fulfilled!

> SD-JWT VCLD credentials are valid SD-JWT VC credentials and all requirements from [@!I-D.ietf-oauth-sd-jwt-vc] apply. Additionally, the requirements listed in this section apply.

Reviewer 4: @steffenschwalm For the microcredentials, see my new comment above.

I agree with @peacekeeper — that's why we have this text in this PR:

> For compatibility with JWT processors, the following registered Claims from [@!RFC7519] and [@!I-D.ietf-oauth-sd-jwt-vc] MUST be used instead of any respective counterpart properties from W3C VCDM or elsewhere:
> 
> * `vct` to represent the type of the Credential.
> * `exp` and `nbf` to represent the validity period of SD-JWT VCLD (i.e., cryptographic signature).
> * `iss` to represent the Credential Issuer.
> * `status` to represent the information to obtain the status of the Credential.
> * `sub` to represent the subject identifier of the Credential.

If that is incomplete or incorrect, let's improve it.

@peacekeeper Right now, we have removed `type` from the LD structure. I'm not sure that works well for processing the LD structure. What about this rule: `vct` and `type` coexist, but `vct` MUST be set to one value in the `type` array. This way we can keep `vct` and `type`.

> SD-JWT VCLD credentials are valid SD-JWT VC credentials and all requirements from [@!I-D.ietf-oauth-sd-jwt-vc] apply. Additionally, the requirements listed in this section apply.

Reviewer 4: In that model, type would be used as before within the `ld` structure, but there would be a vct value "pointing to" one of the type values. Usually, that would be the type value used for requesting the type in the DCQL query. I recognize this is not unambiguous, but it might be an alternative to still provide the type for the JSON-LD part of the processing.

For compatibility with JWT processors, the following registered Claims from [@!RFC7519] and [@!I-D.ietf-oauth-sd-jwt-vc] MUST be used instead of any respective counterpart properties from W3C VCDM or elsewhere:

* `exp` and `nbf` to represent the validity period of SD-JWT VCLD (i.e., cryptographic signature).
* `iss` to represent the Credential Issuer.
* `status` to represent the information to obtain the status of the Credential.
* `sub` to represent the subject identifier of the Credential.

IETF SD-JWT VC is extended with the following claim:

* `ld`: OPTIONAL. Contains a JSON-LD [@!JSON-LD] object in compact form, e.g., [@?VC_DATA].

#### Processing

> The following outlines a suggested non-normative set of processing steps for SD-JWT VCLD:

Reviewer 3: oppose to change to VCLD as SD-JWT VCDM used aby other SDO too

Reviewer 4: Which other SDOs?

Reviewer: can you give links to spec(s) in other SDO(s) that are using the name SD-JWT VCDM? That would be useful to know.

Reviewer 3: ETSI

Reviewer 5: @steffenschwalm Are you saying that an existing European standard currently recognizes something called SD-JWT VCDM which only exists in Pull Requests and in slides, and that this recognition can't be changed to a new name? :)

Reviewer 8: Can be true, but if this spec serves as the base or is being referenced and over here the final decissions have not been made and the name only lives in PRs, IMO other SDOs would have to 'live' with it (lots of ands in that sentence)

Reviewer 3: @nklomp European Standards are mandatory so if OIDF wants recognition for Standard relevant for Europe only it might recognize this fact

Reviewer 4: I checked — it is ETSI 119472-1 and uses "SD-JWT VC DM" to refer to my public repository. That reference needs to be changed. We will reach out to the editor through our OIDF-to-ETSi liaison.

Reviewer 3: @danielfett the document which is public is not the current version. And also if HAIP is moved to VP there`s no need to change for ETSI - as OIDF is no formal body in comparison to ETSI which is mandatory in Europe.

Beside this: As the aim was acc. to the minutes to combine VCDM with SD-JWT (VC) and not only include JSON-LD the name "SD-JWT VCLD" is misleading.  

https://docs.google.com/document/d/1y3milcqMkAHqf4862ANoCA7irg6gNaDI3aNOVaZMoVY/edit?tab=t.0

But if this goal is not in your intention anymore it might be an idea to keep SD_JWT VCLD and to make clear to everybody incl. SDO & EC that the specification here does not fulfil the original scope and agreement (even the one of OIDF with European stakeholders) to make it possible using VCDM in SD-JWT VC.

<{{examples/response/kb_jwt_unsecured.json}}

### SD-JWT VCDM {#sd-jwt_vcdm}

SD-JWT VCDM (Verifiable Credential Data Model) uses the IETF SD-JWT VC [@!I-D.ietf-oauth-sd-jwt-vc] credential format and allows to incorporate existing data models that use Linked Data, e.g., W3C VCDM [@?VC_DATA], while enabling a consistent and uncomplicated approach to selective disclosure.

> 

Reviewer 4: Make it clear that all rules from SD-JWT VC apply:
```suggestion

SD-JWT VCDM credentials are SD-JWT VC credentials and all requirements from [@!I-D.ietf-oauth-sd-jwt-vc] apply. Additionally, the requirements listed in the following apply.

```

Information contained in SD-JWT VCDM credentials can be processed using a JSON-LD [@?JSON-LD] processor after applying SD-JWT VC processing.

When IETF SD-JWT VC is mentioned in this specification, SD-JWT VCDM defined in this section MAY be used.

> 
> Implementers of SD-JWT VCDM MUST use valid values for the `vct` Claim defined in IETF SD-JWT VC [@!I-D.ietf-oauth-sd-jwt-vc].

Reviewer 4: This sentence doesn't add value. I would drop it.
```suggestion

```

For backward compatibility with JWT processors, the following registered JWT claims MUST be used:

* To represent the validity period of SD-JWT VCDM (i.e., cryptographic signature), `exp`/`nbf`/`iat` Claims encoded as a UNIX timestamp (NumericDate) MUST be used.
* `iss` Claim MUST represent the Credential Issuer.
* `status` Claim MUST represent the information to obtain the status of the Credential.
* `sub` Claim MUST represent the subject identifier of the Credential.

IETF SD-JWT VC is extended with the following claims:

* `ld`: OPTIONAL. Contains a JSON-LD [@!JSON-LD] object in compact form, e.g., [@?VC_DATA].

The following outlines a suggested non-normative set of processing steps for SD-JWT VCDM:

> 1. SD-JWT VC Processing:

Reviewer 4: ```suggestion

1. SD-JWT VC Processing:
```

- A receiver (holder or verifier) of an SD-JWT VCDM applies the processing rules outlined in Section 4 of [@!I-D.ietf-oauth-sd-jwt-vc], including verifying signatures, validity periods, status information, etc.
- If the `vct` value is associated with any SD-JWT VC Type Metadata, schema validation of the entire SD-JWT VCDM is performed, including the nested `ld` claim.
- Additionally, trust framework rules are applied, such as ensuring the Credential Issuer is authorized to issue SD-JWT VCDMs for the specified `vct` value.

> 2. Business Logic Processing:

Reviewer 4: ```suggestion
#### Step 2: Business Logic Processing
```

<{{examples/response/kb_jwt_unsecured.json}}

### SD-JWT VCDM {#sd-jwt_vcdm}

SD-JWT VCDM (Verifiable Credential Data Model) extends the IETF SD-JWT VC [@!I-D.ietf-oauth-sd-jwt-vc] credential format and allows to incorporate existing data models that use Linked Data, e.g., W3C VCDM [@?VC_DATA], while enabling a consistent and uncomplicated approach to selective disclosure.

Information contained in SD-JWT VCDM credentials can be processed using a JSON-LD [@?JSON-LD] processor after the SD-JWT VC processing.

When IETF SD-JWT VC is mentioned in this specification, SD-JWT VCDM defined in this section MAY be used.

#### Format

SD-JWT VCDM credentials are valid SD-JWT VC credentials and all requirements from [@!I-D.ietf-oauth-sd-jwt-vc] apply. Additionally, the requirements listed in this section apply.

> For backward compatibility with JWT processors, the following registered JWT claims MUST be used:

Reviewer: I don't think you want to always mandate the presence of `exp`/`nbf`/`iat`/`iss`/`status`/`sub`, do you?

Something more like this maybe (borrowing some words from @Sakurann in https://github.com/openid/oid4vc-haip/pull/147) ? 

```suggestion
For compatibility with JWT processors, the following registered JWT claims are always to be used instead of any respective counterpart properties from W3C VCDM or elsewhere:
```

> * To represent the validity period of SD-JWT VCDM (i.e., cryptographic signature), `exp`/`nbf`/`iat` Claims encoded as a UNIX timestamp (NumericDate) MUST be used.

Reviewer: I will respectfully request `iat` not be characterized as a claim having to do with the validity period of the [SD-]JWT[ VCDM].

Reviewer 4: Will remove iat here.

* `iss` Claim MUST represent the Credential Issuer.
* `status` Claim MUST represent the information to obtain the status of the Credential.

> * `sub` Claim MUST represent the subject identifier of the Credential.

Reviewer: too many MUSTs

- Additionally, trust framework rules are applied, such as ensuring the Credential Issuer is authorized to issue SD-JWT VCDMs for the specified `vct` value.

##### Step 2: Business Logic Processing

- Once the SD-JWT VC is verified and trusted by the SD-JWT VC processor, and if the `ld` claim is present, the receiver extracts the JSON-LD object from the `ld` claim and uses this for the business logic object. If the `ld` claim is not present, the entire SD-JWT VC is considered to represent the business logic object.
- The business logic object is then passed on for further use case-specific processing and validation. The business logic assumes that all security-critical functions (e.g., signature verification, trusted issuer) have already been performed during the previous step. Additional schema validation is applied if provided in the `ld` claim, e.g., to support SHACL schemas. Note that while a `vct` claim is required, SD-JWT VC type metadata resolution and related schema validation is optional in certain cases.

#### Examples

> The following is a non-normative example of an unsecured payload of an SD-JWT VCDM (i.e., before applying the modifications to enable selective disclosure and before adding validity claims). This example is derived from the example of an unsecured payload in Section 3.3 of [@!I-D.ietf-oauth-sd-jwt-vc]:

Reviewer: I am not sure I see what Section 3.3 or SD-JWT VC (link https://datatracker.ietf.org/doc/html/draft-ietf-oauth-sd-jwt-vc-08#section-3.3 and json copied below) has to do with this example here. The "This example is derived from" is only confusing distracting to me. 

This is from 
```
{
  "vct": "https://credentials.example.com/identity_credential",
  "given_name": "John",
  "family_name": "Doe",
  "email": "johndoe@example.com",
  "phone_number": "+1-202-555-0101",
  "address": {
    "street_address": "123 Main St",
    "locality": "Anytown",
    "region": "Anystate",
    "country": "US"
  },
  "birthdate": "1940-01-01",
  "is_over_18": true,
  "is_over_21": true,
  "is_over_65": true
}
```

Reviewer 4: I replaced the example by a newly generated one.

```json
{
   "vct": "https://credentials.example.com/identity_credential",
   "ld":{
      "@context":[
         "https://www.w3.org/ns/credentials/v2",
         "https://w3id.org/citizenship/v3",
      ],
      "type":[
         "VerifiableCredential",
         "PermanentResidentCard"
      ],
      "credentialSubject":{
         "givenName": "John",
         "familyName": "Doe",
         "birthDate": "1978-07-17"
      }
   }
}
```

The following is a non-normative example of the SD-JWT payload after enabling selective disclosure:
```json
{
  "_sd": [

>     "vwUhdFvpylx9Sqi2YNBV1dfVK6lCbhXvkH0nThfKFT0"

Reviewer: so the whole `ld` claim is made selectively disclosable? and nothing else that can be seen here anyway. what is this example showing?

Reviewer 4: Resolved in the new example

- The business logic object is then passed on for further use case-specific processing and validation. The business logic assumes that all security-critical functions (e.g., signature verification, trusted issuer) have already been performed during the previous step. Additional schema validation is applied if provided in the `ld` claim, e.g., to support SHACL schemas. Note that while a `vct` claim is required, SD-JWT VC type metadata resolution and related schema validation is optional in certain cases.

#### Examples

The following is a non-normative example of an unsecured payload of an SD-JWT VCDM (i.e., before applying the modifications to enable selective disclosure and before adding validity claims).

<{{examples/sd_jwt_vcdm/01/user_claims.json}}

The following payload would be used in the SD-JWT after encoding the payload above and enabling selective disclosure on the user-specific claims: 

> <{{examples/sd_jwt_vcdm/01/sd_jwt_payload.json}}

Reviewer: ![Screenshot 2025-04-04 at 3 44 02 PM](https://github.com/user-attachments/assets/387b064f-9945-41fc-af51-6cb03cb454b1)

AFAICT these are the only two of the new files included and that results in only the `credentialSubject` showing up as being made selectively disclosable, which I think does more harm than help. Not having examples at all would be just as useful, maybe more.

If you (we?) want to keep using just these two files, perhaps make `credentialSubject` clear and only redact the sub things? That would at least show some granularity in the selectively disclosable content by haveing an `_sd` with a few items. 

Or using more of the files along with a bit of explanation of each, similar to what you (we?) have done in SD-JWT and SD-JWT VC would make the example more actually meaningful.

Reviewer 4: Showing credentialSubject makes sense. I'll change the example.

Reviewer 4: Döne

Reviewer: Merci

<{{examples/response/kb_jwt_unsecured.json}}

> ### SD-JWT VCDM {#sd-jwt_vcdm}

Reviewer 5: Consider renaming this to SD-JWT VCLD (or similar), to avoid further confusion with W3C VCDM.

Reviewer 4: That makes a lot of sense indeed.

Reviewer: SD-JWT VCLD ain't half bad

Reviewer 9: > as SD-JWT VCDM already recognized by other SDO I strongly oppose against this change

In balance with a strong opposition, "other SDO" is a bit vague.

VCDM has become a non-ambiguous acronym that refers to a particular specification, namely W3C's Verifiable Credential Data Model. If anyone uses "VCDM" for something that doesn't comply with that spec, then they're making a mistake themselves, and certainly shouldn't be used as guidance by others for any naming choice.

Reviewer 5: Even better would be "SD-JWT DCLD", since "VC" also refers to a particular W3C specification :)

- The business logic object is then passed on for further use case-specific processing and validation. The business logic assumes that all security-critical functions (e.g., signature verification, trusted issuer) have already been performed during the previous step. Additional schema validation is applied if provided in the `ld` claim, e.g., to support SHACL schemas. Note that while a `vct` claim is required, SD-JWT VC type metadata resolution and related schema validation is optional in certain cases.

#### Examples

The following is a non-normative example of an unsecured payload of an SD-JWT VCDM (i.e., before applying the modifications to enable selective disclosure and before adding validity claims). This example is derived from the example of an unsecured payload in Section 3.3 of [@!I-D.ietf-oauth-sd-jwt-vc]:

```json
{
   "vct": "https://credentials.example.com/identity_credential",
   "ld":{
      "@context":[
         "https://www.w3.org/ns/credentials/v2",
         "https://w3id.org/citizenship/v3",
      ],

>       "type":[

Reviewer 5: See my early comment about this, I think it's not a good idea to have two ways of expressing the same information (`vct` and `type`): https://github.com/openid/oid4vc-haip/pull/147/files#r1932080113

Author: We have addressed this and removed the `type` property.

* Change Controller: OpenID Foundation Digital Credentials Protocols Working Group - openid-specs-digital-credentials-protocols@lists.openid.net
* Reference: (#openid4vp-scheme) of this specification

## JSON Web Token Claims Registration

> * Claim Name: "ld"

Reviewer: I look forward to recusing myself when this registration request makes it to IANA/DEs.
